src/components/Testimonials/index.jsx

```diff
@@ -30,6 +31,29 @@ const tweets = [
 ];
 
 export function Testimonials() {
+  const [isMobile, setIsMobile] = useState(false);
+  const [visibleTweets, setVisibleTweets] = useState(8);
+  const [showAll, setShowAll] = useState(false); // State to toggle between showing all and limited tweets
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth < 768);
+    };
+
+    handleResize(); // Check initially
```

Comment:
to avoid extra re renders I suggest the following:
```
const [isMobile, setIsMobile] = useState(null) // initiate it as null
```

on the useEffect, avoid calling the function on every render (as [] is empty and you have no way to check)
```
 useEffect(() => {
     // you don't need to declare this next function, you can do it directly below
    // const handleResize = () => {
    //  setIsMobile(window.innerWidth < 768);
    // };

    if (isMobile === null) { // only execute this if isMobile has not been set yet
     setIsMobile(window.innerWidth < 768);
     window.addEventListener('resize', handleResize);
    }
    return () => window.removeEventListener('resize', handleResize);
}, [isMobile]);
```

Reply:
hey, 
Do you mean handleResize inside event listener?
` window.addEventListener('resize', () => {
        setIsMobile(window.innerWidth < 768);
    }); `
But that will create a problem while removing the event listener as it creates different references.


Also the function runs only on inital mount unless we use isMobile in the dependency array